command/agent.go

```diff
@@ -1219,3 +1212,169 @@ func (c *AgentCommand) handleQuit(enabled bool) http.Handler {
 		close(c.ShutdownCh)
 	})
 }
+
+// newLogger creates a logger based on parsed config field on the Agent Command struct.
+func (c *AgentCommand) newLogger() (log.InterceptLogger, error) {
+	if c.config == nil {
+		return nil, fmt.Errorf("cannot create logger, no config")
+	}
+
+	var errors error
+
+	// Parse all the log related config
+	logLevel, err := logging.ParseLogLevel(c.config.LogLevel)
+	if err != nil {
+		errors = multierror.Append(errors, err)
+	}
+
+	logFormat, err := logging.ParseLogFormat(c.config.LogFormat)
+	if err != nil {
+		errors = multierror.Append(errors, err)
+	}
+
+	logRotateDuration, err := parseutil.ParseDurationSecond(c.config.LogRotateDuration)
+	if err != nil {
+		errors = multierror.Append(errors, err)
+	}
+
+	logRotateBytes, err := parseutil.ParseInt(c.config.LogRotateBytes)
+	if err != nil {
+		errors = multierror.Append(errors, err)
+	}
+
+	logRotateMaxFiles, err := parseutil.ParseInt(c.config.LogRotateMaxFiles)
+	if err != nil {
+		errors = multierror.Append(errors, err)
+	}
+
+	if errors != nil {
+		return nil, errors
+	}
+
+	logCfg := &logging.LogConfig{
+		Name:              "vault-agent",
+		LogLevel:          logLevel,
+		LogFormat:         logFormat,
+		LogFilePath:       c.config.LogFile,
+		LogRotateDuration: logRotateDuration,
+		LogRotateBytes:    int(logRotateBytes),
+		LogRotateMaxFiles: int(logRotateMaxFiles),
+	}
+
+	l, err := logging.Setup(logCfg, c.logWriter)
+	if err != nil {
+		return nil, err
+	}
+
+	return l, nil
+}
+
+// loadConfig attempts to generate an Agent config from the file(s) specified.
+func (c *AgentCommand) loadConfig(paths []string) (*agentConfig.Config, error) {
+	var errors error
+	cfg := agentConfig.NewConfig()
+
+	for _, configPath := range paths {
+		configFromPath, err := agentConfig.LoadConfig(configPath)
+		if err != nil {
+			errors = multierror.Append(errors, fmt.Errorf("error loading configuration from %s: %w", configPath, err))
+		} else {
+			cfg = cfg.Merge(configFromPath)
+		}
+	}
+
+	if errors != nil {
+		return nil, errors
+	}
+
+	if err := cfg.ValidateConfig(); err != nil {
+		return nil, fmt.Errorf("error validating configuration: %w", err)
+	}
+
+	return cfg, nil
+}
+
+// reloadConfig will attempt to reload the config from file(s) and adjust certain
```

Comment:
I think it might be worth us pointing out in this comment that this is triggered by SIGHUP, obvious to us now, but potentially less obvious to us later.